docs/developer-guide/api-overview.md

```diff
@@ -184,45 +186,30 @@ Styx core also add snippets of information like sender IP address, and so on.
 
 ### Http Handler Interface
 
-Http Handler interface forms a basis for Styx admin interfaces and routing objects. 
-As opposed to `HttpInterceptor`s, which just pass the messages down the pipeline,
-the `HttpHandler` is meant to *consume* the HTTP request. It is a similarly simple
-interface:
+A `HttpHandler` interface is the basis for Styx admin interfaces and routing objects. 
+It *consumes* the HTTP requests and asynchronously responds with HTTP responses. 
+It has a simple interface:
 
 ```java
     public interface HttpHandler {
-        StyxObservable<HttpResponse> handle(HttpRequest request, HttpInterceptor.Context context);
+        Eventual<LiveHttpResponse> handle(LiveHttpRequest request, HttpInterceptor.Context context);
     }
 ```
 
-It asynchronously processes the request, and returns the response within a context of 
-`StyxObservable`. It is given the `HttpInterceptor.Context` as a second argument, so that it is able
-to access the request context properties.
-
-Notice the absence of `Chain`. Therefore it is not able to proceed the message any further.
+It accepts a live HTTP request as its first argument, and eventually responds with live HTTP reponse. 
+The second argument is a `HttpInterceptor.Context`. The request context contains contextual
+information such as the sender of the request and whether it was delivered over a secure protocol or not.
 
 As with `HttpInterceptor` implementations, the `handle` method must never block. Blocking the
 thread will block the Styx event loop.  
 
 
-### Styx Observable
-
-Conceptually similar to Futures, `StyxObservable` is a data type that 
-facilitates asynchronous event handling, modelled after Rx 
-[observables](http://reactivex.io/documentation/observable.html).
-
-However `rx.Observable` is a very generic reactive stream abstraction. 
-The `StyxObservable` is an observable that has been adapted 
-for the specific Styx use case, which is of processing live network data 
-streams. 
+### Eventual
 
-Another key difference between the two observables is that `StyxObservable`
-does not have a `subscribe` method. More precisely it has been hidden 
-to prevent 3rd party extensions from subscribing to live data streams. 
-This is a privileged operation that for the reliable operation is exclusive 
-for Styx core only.  
+Styx `Eventual` class is an envelope for a value that is available some time in future.
```

Comment:
"Styx `Eventual` class" -> "The `Eventual` class"

"value that is available some time in future" -> "value that will be available at some time in the future"